mobile/android/android-emulator-services.ts

```diff
@@ -120,29 +147,105 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 		return timeout * 1000;
 	}
 
+
+	private getRunningEmulatorId(image: string): IFuture<string> {
+		return ((): string => {
+			var runningEmulators = this.getRunningEmulators().wait();
+			if(runningEmulators.length === 0) {
+				return "";
+			}
+
+			// if we get here, there's at least one running emulator
+			var getNameFunction = options.geny ? this.getNameFromGenymotionEmulatorId : this.getNameFromNativeEmulatorId;
+			var emulatorId = _(runningEmulators).find(emulator => getNameFunction.apply(this, [emulator]).wait() === image);
+
+			return emulatorId;
+		}).future<string>()();
+	}
+
+	private getNameFromGenymotionEmulatorId(emulatorId: string): IFuture<string> {
+		return (() => {
+			var modelOutputLines: string = this.$childProcess.execFile(this.$staticConfig.adbFilePath, ["-s", emulatorId, "shell", "getprop", "ro.product.model"]).wait();
+			this.$logger.trace(modelOutputLines);
+			var model = (<string>_.first(modelOutputLines.split(os.EOL))).trim();
+			return model;
+		}).future<string>()();
+	}
+
+	private getNameFromNativeEmulatorId(emulatorId: string): IFuture<string> {
```

Comment:
What about naming this function similar to "getNameFromSdkEmulatorId" -> both are native emulators.